api/types/plugin.go

```diff
@@ -97,6 +101,13 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if err := p.Credentials.GetOauth2AccessToken().CheckAndSetDefaults(); err != nil {
 			return trace.Wrap(err)
 		}
+	case *PluginSpecV1_Opsgenie:
+		if settings.Opsgenie == nil {
+			return trace.BadParameter("settings must be set")
```

Comment:
Should we add some `opsgenie` reference to the error message?
```suggestion
			return trace.BadParameter("opsgenie settings must be set")
```